app/src/test/java/com/example/memeitupapp/listmemetest/listmodeltest/ListModelTest.kt

```diff
@@ -22,31 +24,37 @@ class ListModelTest {
     @get:Rule
     val instantTaskExecutorRule = InstantTaskExecutorRule()
 
-    private val memeService: MemeService = mock()
+    private val mockedGetMemesUseCase: GetMemesUseCase = mock()
+    private val mockedGetMemesFromDataBaseUseCase: GetMemesFromDataBaseUseCase = mock()
+    private val mockedUpdateMemesDataBaseUseCase: UpdateMemesDataBaseUseCase = mock()
     private lateinit var listMemesModel: ListMemesContract.Model
 
     @Before
     fun setup() {
-        listMemesModel = ListMemesModel(memeService)
+        listMemesModel = ListMemesModel(mockedGetMemesUseCase, mockedUpdateMemesDataBaseUseCase, mockedGetMemesFromDataBaseUseCase)
     }
 
     @Test
-    fun `on get memes for list successfully`() {
+    fun `on GetMemesUseCase gets data from API, update data base`() {
         val mockedMemesResult: Result.Success<List<Meme>> = mock()
-        whenever(memeService.getMemesFromApi()).thenReturn(mockedMemesResult)
-
-        runBlocking { listMemesModel.getMemes() }
+        whenever(mockedGetMemesUseCase.invoke()).thenReturn(mockedMemesResult)
 
         assertEquals(mockedMemesResult, listMemesModel.getMemes())
+
+        verify(mockedGetMemesUseCase).invoke()
+        verify(mockedUpdateMemesDataBaseUseCase).invoke(mockedMemesResult.data)
     }
 
     @Test
-    fun `on get memes for list with error connection`() {
-        val mockedMemesResult: Result.Failure = mock()
-        whenever(memeService.getMemesFromApi()).thenReturn(mockedMemesResult)
+    fun `on GetMemesUseCase fails, invoke GetMemesFromDataBaseUseCase`() {
+        val mockedMemesResultFailure: Result.Failure = mock()
```

Comment:
you can omit this variable because you don't do any assert with it

Reply:
You are right, deleted!